src/modules/tradeQuote/hooks/useQuoteParams.ts

```diff
@@ -1,22 +1,34 @@
 import { useAtomValue } from 'jotai'
 import { useMemo } from 'react'
 
+import { OrderKind } from '@cowprotocol/cow-sdk'
+import { CurrencyAmount } from '@uniswap/sdk-core'
+
 import { NATIVE_CURRENCY_BUY_ADDRESS } from 'legacy/constants'
 
+import { useEnoughBalanceAndAllowance } from 'modules/tokens'
 import { useDerivedTradeState } from 'modules/trade/hooks/useDerivedTradeState'
 import { useWalletInfo } from 'modules/wallet'
 
+import { getPriceQuality } from 'api/gnosisProtocol/api'
 import { getAddress } from 'utils/getAddress'
 
 import { tradeQuoteParamsAtom } from '../state/tradeQuoteParamsAtom'
 
 export function useQuoteParams() {
   const { chainId, account } = useWalletInfo()
   const { state } = useDerivedTradeState()
+
   const { amount } = useAtomValue(tradeQuoteParamsAtom)
 
   const { inputCurrency, outputCurrency, orderKind } = state || {}
 
+  const currency = orderKind === OrderKind.SELL ? inputCurrency : outputCurrency
+  const enoughBalance = useEnoughBalanceAndAllowance({
+    account,
+    amount: (currency && amount && CurrencyAmount.fromRawAmount(currency, amount)) || undefined,
+  })
```

Comment:
Same question here regarding the quote amount and sell vs buy token.

Reply:
Fixed 7d3de5996b5b04958f000494b16afacf163ff004

Reply:
Fixed 7d3de5996b5b04958f000494b16afacf163ff004